spec/queries/keywords_query_spec.rb

```diff
@@ -26,19 +26,132 @@
           expect(keywords).to be_empty
         end
       end
+    end
 
+    context 'when querying with adwords_url_contains' do
       context 'given filter_params is nil' do
         it 'returns Keyword scope' do
           keywords = described_class.new(Keyword, nil).call
 
           expect(keywords).to eq Keyword
         end
       end
+
+      context 'when querying with no parameters' do
+        it 'returns Keyword scope' do
+          keywords = described_class.new(Keyword, {}).call
+
+          expect(keywords).to eq Keyword
+        end
+      end
+
+      context 'when there are urls containing the word vpn' do
+        it 'returns 2 keywords' do
+          ads_top_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
+          Fabricate.times(2, :keyword, ads_top_urls: ads_top_urls)
```

Comment:
When testing something like this, let try to add another "unrelated" keyword to make sure it returns only 2
```suggestion
          ads_top_urls = ['https://www.thetopvpn.com', 'https://www.nordvpn.com', 'https://www.vnexpress.net']
          Fabricate.times(2, :keyword, ads_top_urls: ads_top_urls)     
          Fabricate(:keyword, ads_top_urls: ['https://www.vpe.com'])
```

Reply:
And in the contexts order, this happy case should come before 2 above contexts

Reply:
Moved the above 2 contexts out of this block as they were not related.